dandi/download.py

```diff
@@ -501,3 +502,68 @@ def _download_file(
         os.utime(path, (time.time(), ensure_datetime(mtime).timestamp()))
 
     yield {"status": "done"}
+
+
+class DownloadDirectory:
+    def __init__(self, filepath, filehash):
+        #: The path to which to save the file after downloading
+        self.filepath = Path(filepath)
+        #: The expected hash of the downloaded data
+        self.filehash = filehash
+        #: The working directory in which downloaded data will be temporarily
+        #: stored
+        self.dirpath = append_ext(self.filepath, ".download")
```

Comment:
unlikely we would ever have "legit" `X` and `X.download` files in the archive, but let's make it `.dandidownload` to reduce possibility of collision even further?

Reply:
Done.